tests/func/test_api.py

```diff
@@ -8,9 +8,9 @@
 from dvc.main import main
 from dvc.path_info import URLInfo
 from dvc.utils.fs import remove
-from tests.remotes import GCP, HDFS, OSS, S3, SSH, Azure, Local
+from tests.remotes import GCP, HDFS, OSS, S3, SSH, Azure, GDrive, Local
 
-remote_params = [S3, GCP, Azure, OSS, SSH, HDFS]
+remote_params = [S3, GCP, Azure, GDrive, OSS, SSH, HDFS]
```

Comment:
we have a lot of tests for functions besides open here - do we expect all of them to pass? we'll need to implement more stuff then

Reply:
@shcheklein All the tests should work, there is nothing new required from GdriveRemotes. There might be some issues with Gdrive itself though (creds or something), that differentiates it from other remotes.

Reply:
Even it works out of the box, some things do not make much sense - e.g. what will `dvc.api.url` return for the object? Some URL that no tool (except DVC) can work with?

Re credentials - should be fine (we use an env var to set them, so I would expect external repo to pick it up as well). As I mentioned though, GDrive remote requires some special setup (e.g. it does not create a directory for you).

Reply:
regarding `get_url`, I think that's how the value should look like:

<img width="830" alt="Screen Shot 2020-06-04 at 5 00 38 PM" src="https://user-images.githubusercontent.com/3659196/83822240-5418e400-a685-11ea-83e1-f3db1e3efa44.png">

(`id=` is the file ID that we can get via API)

Reply:
@shcheklein is this comment outdated since rebasing on master? Or does `tests/remotes.py` still need updating?

Reply:
@casperdcl primarily gdrive logic should be updated I think, the one that generates URLs - right now they probably look like `gdrive://root/....` which is not very meaningful - no tools understand `gdrive://` URLs . Also, it probably returns path not ID, which is potentially not unique.

Ideally we would have an URL like on the image above (you can click and download in browser). Or it might be somethings else.

The bottom line here is that `get_url` for GDrive needs probably some research on its own and can be a separate PR, and for now we probably should throw NotImplemented.

Unless you guys see some easy and fast implementation.

Reply:
I think you're talking about `api.get_url` rather than `tests.remotes.GDrive.get_url`. Pretty sure the latter is *meant* to work as-is after @efiop's changes but the tests are still failing requiring 2FA/confirmation codes.

Reply:
@casperdcl You need to `create_dir`, `get_url` is not returning an already created dir.

Reply:
Hmm. Any reason we can't move `create_dir` to within `get_url`?

Reply:
@casperdcl Could do that, sure. Will require some work to make it do that without first passing it a dvc instance.

Reply:
well the only other way I see is to rework the current pytest fixture (currently it hides the `GDrive` object so there's no way to call `create_dir`)

Reply:
@casperdcl Both work. Choose the easiest one, we'll be revisiting this later anyway.

Reply:
tried a simple change along the same lines as `setup_remote` but still failing with the same credentials error https://travis-ci.com/github/iterative/dvc/jobs/348374056#L7952

Reply:
@casperdcl does it work locally?